woodwork/table_accessor.py

```diff
@@ -38,24 +38,53 @@
 dd = import_or_none('dask.dataframe')
 ks = import_or_none('databricks.koalas')
 
-
+from typing import Optional, Union
+from woodwork.logical_types import LogicalType
 class WoodworkTableAccessor:
     def __init__(self, dataframe):
         self._dataframe_weakref = weakref.ref(dataframe)
         self._schema = None
 
-    def init(self,
-             index=None,
-             time_index=None,
-             logical_types=None,
-             already_sorted=False,
-             schema=None,
-             validate=True,
-             use_standard_tags=True,
-             **kwargs):
-        """Initializes Woodwork typing information for a DataFrame.
+    def init(self, schema=None, **kwargs):
+        """Initializes Woodwork typing information for a DataFrame with only keyword arguments.
+        logical type inference is performed on columns with unspecified logical types"""
+        if schema is not None:
+            raise ValueError("call init_with_partial_schema or init_with_full_schema to pass a schema")
+        self.init_with_partial_schema(schema=None, **kwargs)
+   
+    def init_with_full_schema(self, schema: TableSchema, validate: bool = True) -> None:
+        """Initializes Woodwork typing information for a DataFrame with a complete schema"""
+        if validate:
+            _check_schema(self._dataframe, schema)
+            _check_unique_column_names(self._dataframe)
+        self._schema = schema
+
+    def init_with_partial_schema(self,
+                                 schema: Optional[TableSchema]=None,
+                                 index: Optional[str]=None,
+                                 time_index: Optional[str]=None,
+                                 logical_types: Optional[dict[str, LogicalType]]=None,
+                                 already_sorted: Optional[bool]=False,
+                                 name: Optional[str]=None,
+                                 semantic_tags: Optional[dict[str, Union[str, list[str], set[str]]]]=None,
+                                 table_metadata: Optional[dict]=None,
+                                 column_metadata: Optional[dict[str, dict]]=None,
```

Comment:
These params that can take in a dictionary where they keys are column names should allow the column names to be anything that can be a pandas column name (so it allows anything hashable like numbers). Something I think could be incredibly useful for us would be to create a `column_name` type alias for these params. So we could say `Optional[dict[column_name, dict]]` and `column_name` would be something like `Union[str, number, tuple]`

@davesque is there a way with python's types to make some sort of type alias? (or maybe the question should be: Is that an advisable thing to do even if it's possible)

Reply:
I've made an alias for ColumnName